client/lxd.go

```diff
@@ -492,3 +492,48 @@ func (r *ProtocolLXD) getUnderlyingHTTPTransport() (*http.Transport, error) {
 		return nil, fmt.Errorf("Unexpected http.Transport type, %T", r)
 	}
 }
+
+func (r *ProtocolLXD) optimizeSrcImage(source ImageServer, image api.Image, instSrc *api.InstanceSource) (isSameServer bool, info *ConnectionInfo, err error) {
```

Comment:
This function name isn't that useful to me. It isn't actually optimizing an image (what would that mean anyway?).

How about `getSourceImageConnectionInfo()`?

Also, I'm not sure we need the `isSameServer` return value.
It looks like if the function returns `nil, nil` then its the same server, otherwise it will return `*ConnectionInfo, nil`, or err will be non-nil.

So I think we can simplify the return values, and explain them in a doc block for this function.
Also we should include in the docblock that the instSrc is modified.